Source/SwiftLintFramework/Models/Configuration.swift

```diff
@@ -10,6 +10,12 @@ import Foundation
 import SourceKittenFramework
 
 public struct Configuration: Equatable {
+    // Represents how a Configuration object can be configured with regards to rules.
+    public enum RulesMode {
```

Comment:
1. I also thought of `RulesConfiguration`, but it'd be kinda confusing as `RuleConfiguration` is already used. Other that than I can't find of a better name either :/
2. +1 for `Set<String>`, will make the change. We can make the duplicate validation earlier in `Configuration+Parsing`
3. Maybe `overrideDefaults`?